proposals/GlobalUsingDirective.md

```diff
@@ -2,13 +2,194 @@
+            * If `K` is zero and the namespace declaration contains an *extern_alias_directive* or *using_alias_directive* that associates the name `I` with an imported namespace or type, **or any namespace declaration for `N` in the program contains a *global_using_alias_directive* that associates the name `I` with an imported namespace or type,** then the *namespace_or_type_name* refers to that namespace or type.
+            * Otherwise, if the namespaces and type declarations imported by the *using_namespace_directive*s and *using_alias_directive*s of the namespace declaration contain exactly one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* refers to that type constructed with the given type arguments.
+            * Otherwise, if the namespaces and type declarations imported by the *using_namespace_directive*s and *using_alias_directive*s of the namespace declaration contain more than one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* is ambiguous and an error occurs.
+            * **Otherwise, if the namespaces and type declarations imported by the *global_using_namespace_directive*s and *global_using_static_directive*s of any namespace declaration for `N` in the program contain exactly one accessible type having name `I` and `K` type parameters, then the *namespace_or_type_name* refers to that type constructed with the given type arguments.**
+            * **Otherwise, if the namespaces and types imported by the *global_using_namespace_directive*s and *global_using_static_directive*s of any namespace declaration for `N` in the program contain more than one accessible typehaving name `I` and `K` type parameters, then the *namespace_or_type_name* is ambiguous and an error occurs.**
+    * Otherwise, the *namespace_or_type_name* is undefined and a compile-time error occurs.
+
+# Simple names
+https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#simple-names
+
+Changes are made to the *simple_name* evaluation rules as follows.
+
+This is the relevant bullet point with proposed additions (which are **in bold**):
+*  Otherwise, for each namespace `N`, starting with the namespace in which the *simple_name* occurs, continuing with each enclosing namespace (if any), and ending with the global namespace, the following steps are evaluated until an entity is located:
+   *  If `K` is zero and `I` is the name of a namespace in `N`, then:
+      * If the location where the *simple_name* occurs is enclosed by a namespace declaration for `N` and the namespace declaration contains an *extern_alias_directive* or *using_alias_directive* that associates the name `I` with a namespace or type, **or any namespace declaration for `N` in the program contains a *global_using_alias_directive* that associates the name `I` with a namespace or type,** then the *simple_name* is ambiguous and a compile-time error occurs.
```

Comment:
Continued questions about `N` containing a *global_using_directive* or *global_using_alias_directive*

Reply:
Same thing

---
In reply to: [589639783](https://github.com/dotnet/csharplang/pull/4499#discussion_r589639783) [](ancestors = 589639783)